sway/tree/view.c

```diff
@@ -73,7 +74,46 @@ void view_set_activated(struct sway_view *view, bool activated) {
 	}
 }
 
+void view_set_fullscreen(struct sway_view *view, bool fullscreen) {
+	if (view->is_fullscreen == fullscreen) {
+		return;
+	}
+
+	struct sway_container *container = container_parent(view->swayc, C_OUTPUT);
+	struct sway_output *output = container->sway_output;
+	struct sway_container *workspace = container_parent(view->swayc, C_WORKSPACE);
+
+	if (view->impl->set_fullscreen) {
+		view->impl->set_fullscreen(view, fullscreen);
+	}
+
+	if (fullscreen) {
+		view->swayc->saved_x = view->swayc->x;
+		view->swayc->saved_y = view->swayc->y;
+		view->saved_width = view->width;
+		view->saved_height = view->height;
+		view_configure(view, 0, 0, output->wlr_output->width, output->wlr_output->height);
+		workspace->fullscreen = view;
+	} else {
+		view_configure(view, view->swayc->saved_x, view->swayc->saved_y,
+				view->saved_width, view->saved_height);
+		workspace->fullscreen = NULL;
+	}
+
+	view->is_fullscreen = fullscreen;
```

Comment:
I don't think this is so simple. You have to wait until the surface commits with a buffer the size of the screen before you can actually set it to be fullscreen. Otherwise you will flash partially rendered content.

Reply:
We're not doing the "frame perfect" milestone yet.